### PR TITLE
EuroToken III: Add Trust Scores to EuroToken

### PR DESCRIPTION
**Pull request by pietertolsma**
This PR adds the concept of trust to the EuroToken app.

Check out [our issue](https://github.com/Tribler/tribler/issues/6785).

For a full explainer on this PR, check out eurotoken/README.md

### CODE REVIEW

**Comment by Fastjur** [comment]
Attached debug APK: https://drive.google.com/file/d/1-qag_BBYBweuw9ENgHTe0t8p8RdR2nkX/view?usp=sharing

File: eurotoken/README.MD [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# EuroToken - A Digital Euro
+_TODO: Write out in more detail about basics of EuroToken_
```
Comment (InvictusRMC) [inline eurotoken/README.MD:2]: TODO in readme.

File: eurotoken/README.MD [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#### Requirements for solution
+To adhere to the standard of EuroToken, the solution for double spending has to comply to the following requirements:
+* Fully offline capable
+* Completely distributed
+* Permissionless
+* Pseudo-anonymous
+* Independent of other authorities, such as banks
+
+### Double Spending mitigation
+To counter double spending, EuroToken attempts to create a [web-of-trust](https://en.wikipedia.org/wiki/Web-of-trust) to help verify the authenticity of transactions and users.
```
Comment (InvictusRMC) [inline eurotoken/README.MD:32]: This page does not exist, you probable mean: https://en.wikipedia.org/wiki/Web_of_trust

File: eurotoken/src/main/java/nl/tudelft/trustchain/eurotoken/community/EuroTokenCommunity.kt [hunk 3]
```diff
@@ -71,13 +118,96 @@ class EuroTokenCommunity(
...
+            // Generate [num] addresses if in demo mode
+            addresses.addAll(generatePublicKeys(num))
+        } else {
+            // Get all addresses of the last [num] incoming transactions
+            addresses.addAll(transactionRepository.getTransactions(num).map { transaction: Transaction ->
+                transaction.sender.toString()
+            })
+        }
+
+        val payload = TransactionsPayload("1", addresses.joinToString(separator = ",").toByteArray())
```
Comment (InvictusRMC) [inline eurotoken/src/main/java/nl/tudelft/trustchain/eurotoken/community/EuroTokenCommunity.kt:185]: Probably better to use the actual message ID const you defined earlier.

**Review by InvictusRMC — CHANGES_REQUESTED** [review]
Some very minor changes requested. See comments

**Comment by Fastjur** [comment]
> In addition to the comments by @InvictusRMC, please squash your commit history into one commit.

Yes we can squash merge once all the reviewing is done :)

File: eurotoken/src/main/java/nl/tudelft/trustchain/eurotoken/community/EuroTokenCommunity.kt (hunk 3 above)
Comment (pietertolsma) [reply eurotoken/src/main/java/nl/tudelft/trustchain/eurotoken/community/EuroTokenCommunity.kt]: Done!

**Review by devos50 — CHANGES_REQUESTED** [review]
It seems that your PR is changing the `kotlin-ipv8` submodule pointer to an older commit. Please remove this from your PR (the submodule pointer should not change, unless you have merged changes in the `kotlin-ipv8` submodule).

**Comment by Fastjur** [comment]
> It seems that your PR is changing the `kotlin-ipv8` submodule pointer to an older commit. Please remove this from your PR (the submodule pointer should not change, unless you have merged changes in the `kotlin-ipv8` submodule).

Good catch, thanks. Not sure how that happened, I'll look into it!

**Review by InvictusRMC — APPROVED** [review]

**Comment by InvictusRMC** [comment]
Hi team, your code has been approved! Please squash your commit history.

**Review by devos50 — APPROVED** [review]

**Comment by pietertolsma** [comment]
Great! Can it be squashed + merge by someone from the team with write access?

**Pull request merged**